cloud/util/util.go

```
@@ -317,14 +321,14 @@ func InitClientsAndRegion(ctx context.Context, client client.Client, defaultRegi
 // CreateClientProviderFromClusterIdentity creates scope.ClientProvider from Cluster Identity
 func CreateClientProviderFromClusterIdentity(ctx context.Context, client client.Client, namespace string, defaultRegion string, clusterAccessor scope.OCIClusterAccessor, identityRef *corev1.ObjectReference) (*scope.ClientProvider, error) {
 	identity, err := GetClusterIdentityFromRef(ctx, client, namespace, identityRef)
-	if err != nil {
+	if err != nil || identity == nil {
 		return nil, err
```

If `err` is `nil` and `identity` is `nil` won't this just `return nil, nil`? Should we return an error with a message that `identity` is `nil` or something?

---

I've removed the identity check as this case should never occur and added no lint:nilaway to suppress the issue in the lint checker